File: app/src/main/java/com/uwetrottmann/seriesguide/extensionexample/app/MainActivity.java
package com.uwetrottmann.seriesguide.extensionexample.app;

import android.content.Intent;
import android.os.Bundle;

import com.battlelancer.seriesguide.api.Intents;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.buttonMainViewShow).setOnClickListener(v -> {
            // view show in SeriesGuide
            Intent intent = Intents.buildViewShowIntent(248835);
            if (intent.resolveActivity(getPackageManager()) != null) {
                startActivity(intent);
            }
        });

        findViewById(R.id.buttonMainViewEpisode).setOnClickListener(v -> {
            // view episode in SeriesGuide
            Intent intent = Intents.buildViewEpisodeIntent(248835, 4164086);
            if (intent.resolveActivity(getPackageManager()) != null) {
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/uwetrottmann/seriesguide/extensionexample/app/MainActivity.kt
package com.uwetrottmann.seriesguide.extensionexample.app

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.battlelancer.seriesguide.api.Intents
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Find the views by their IDs
        val editText: EditText = findViewById(R.id.editText)
        val saveButton: Button = findViewById(R.id.saveButton)

        // Set a click listener for the Save button
        saveButton.setOnClickListener {
            // Get the text from the EditText
            val text = editText.text.toString()

            // Save the text or show it in a Toast for now
            if (text.isNotEmpty()) {
                // For example, show a Toast as feedback
                Toast.makeText(this, "Saved: $text", Toast.LENGTH_SHORT).show()

                // Here you could save it in SharedPreferences, database, etc.
            } else {
                // Handle empty input
                Toast.makeText(this, "Please enter some text", Toast.LENGTH_SHORT).show()
            }
        }
    }

}
